support/traits/src/lib.rs

```diff
@@ -49,6 +50,87 @@ pub trait ProvidersInterface {
+pub trait MutateProvidersInterface {
+    /// The type which can be used to identify accounts.
+    type AccountId: Parameter + Member + MaybeSerializeDeserialize + Debug + Ord + MaxEncodedLen;
+    /// The type which represents a registered Provider.
+    type Provider: Parameter + Member + MaybeSerializeDeserialize + Debug + Ord + MaxEncodedLen;
+    /// Data type for the measurement of storage size
+    type StorageData: Parameter
+        + Member
+        + MaybeSerializeDeserialize
+        + Default
+        + MaybeDisplay
+        + AtLeast32BitUnsigned
+        + Copy
+        + MaxEncodedLen
+        + HasCompact;
+    /// The type of ID that uniquely identifies a Merkle Trie Holder (BSPs/Buckets) from an AccountId
+    type BucketId: Parameter
+        + Member
+        + MaybeSerializeDeserialize
+        + Debug
+        + MaybeDisplay
+        + SimpleBitOps
+        + Ord
+        + Default
+        + Copy
+        + CheckEqual
+        + AsRef<[u8]>
+        + AsMut<[u8]>
+        + MaxEncodedLen
+        + FullCodec;
+    /// The type of the Merkle Patricia Root of the storage trie for BSPs and MSPs' buckets (a hash).
+    type MerklePatriciaRoot: Parameter
+        + Member
+        + MaybeSerializeDeserialize
+        + Debug
+        + MaybeDisplay
+        + SimpleBitOps
+        + Ord
+        + Default
+        + Copy
+        + CheckEqual
+        + AsRef<[u8]>
+        + AsMut<[u8]>
+        + MaxEncodedLen
+        + FullCodec;
+
+    /// Increase the used data of a Storage Provider (generic, MSP or BSP).
+    fn increase_data_used(who: &Self::AccountId, delta: Self::StorageData) -> DispatchResult;
+
+    /// Decrease the used data of a Storage Provider (generic, MSP or BSP).
+    fn decrease_data_used(who: &Self::AccountId, delta: Self::StorageData) -> DispatchResult;
+
+    /// Add a new Bucket as a Provider
+    fn add_bucket(
+        msp_id: Self::Provider,
+        user_id: Self::AccountId,
+        bucket_id: Self::BucketId,
+        bucket_root: Self::MerklePatriciaRoot,
+    ) -> DispatchResult;
+
+    /// Change the root of a bucket
+    fn change_root_bucket(
+        bucket_id: Self::BucketId,
+        new_root: Self::MerklePatriciaRoot,
+    ) -> DispatchResult;
+
+    /// Change the root of a BSP
+    fn change_root_bsp(
+        bsp_id: Self::Provider,
+        new_root: Self::MerklePatriciaRoot,
+    ) -> DispatchResult;
+
+    /// Remove a root from a bucket of a MSP, removing the whole bucket from storage
+    fn remove_root_bucket(bucket_id: Self::BucketId) -> DispatchResult;
+
+    /// Remove a root from a BSP. It will remove the whole BSP from storage, so it should only be called when the BSP is being removed.
+    /// todo!("If the only way to remove a BSP is by this pallet (bsp_sign_off), then is this function actually needed?")
+    fn remove_root_bsp(who: &Self::AccountId) -> DispatchResult;
+}
+
```

Comment:
I don't know if you saw on Slack but Facu wanted to discuss this as this might not be needed. I'll ping you on the thread again

Reply:
As discussed, this trait method is removed.